src/scripts/model/master.js

```diff
@@ -2,204 +2,205 @@
  *
  */
-function MasterModel(opts) {
+class MasterModel extends EventEmitter {
 
-    if (!(this instanceof MasterModel)) {
-        return new MasterModel(opts);
-    }
+    /**
+     *
+     * @param {Object} opts shared options Object
+     *
+     */
+    constructor(opts) {
 
-    this.opts = assign({
-        interval: 40000,
-        enableFetch: true,
-        enablePolling: true,
-        path: ''
-    }, opts);
+        super();
 
-    this.instance = false;
-    this.dataCache = {};
-    this.dataFetch = null;
+        this.opts = assign({
+            interval: 40000,
+            enableFetch: true,
+            enablePolling: true,
+            path: ''
+        }, opts);
 
-    EventEmitter.call(this);
+        this.instance = false;
+        this.dataCache = {};
+        this.dataFetch = null;
 
-}
+    }
+
+    /**
+     *
+     * @method init
+     * @desc initiates master model, begins initial data fetch
+     * @returns {Object} this
+     *
+     */
+    init() {
 
-util.inherits(MasterModel, EventEmitter);
+        const { instance } = this;
+        const { enableFetch } = this.opts;
 
-/**
- *
- * @method init
- * @description initiates master model, begins initial data fetch
- * @returns {Object} this
- *
- */
-MasterModel.prototype.init = function() {
+        if (enableFetch && !instance) {
 
-    let { instance } = this;
-    let { enableFetch } = this.opts;
+            this.instance = true;
+            this.fetch();
 
-    if (enableFetch && !instance) {
+        }
 
-        this.instance = true;
-        this.fetch();
+        return this;
 
     }
 
-    return this;
+    /**
+     *
+     * @method fetch
+     * @desc makes a GET request to specified path, emits data event, expecting JSON by default
+     * @returns {Object} this
+     *
+     */
+    fetch() {
+
+        const { opts } = this;
+        const { path, enablePolling, interval } = opts;
 
-};
+        utils.log('fetch master model data');
 
-/**
- *
- * @method fetch
- * @description makes a GET request to specified path, emits data event, expecting JSON by default
- * @returns {Object} this
- *
- */
-MasterModel.prototype.fetch = function() {
+        /**
+         *
+         * @function handleData
+         * @param {Object} response
+         * @returns {Object} response JSON
+         * @desc handles response and returns JSON if successful
+         *
+         */
+        function handleData(response) {
 
-    let { opts } = this;
-    let { path, enablePolling, interval } = opts;
+            if (response.status >= 400) {
+                throw response.status;
+            }
 
-    utils.log('fetch master model data');
+            try {
 
-    /**
-     *
-     * @function handleData
-     * @param {Object} response
-     * @returns {Object} response JSON
-     * @description handles response and returns JSON if successful
-     *
-     */
-    function handleData(response) {
+                return response.json();
```

Comment:
Thoughts on adding a fallback here?

For example, this is what I set up for the SDK `HttpClient`.

```js
return response.clone().json().catch(() => {

    return response.text();

}).then((data) => {

    return Promise.resolve({
        status,
        data,
        headers
    });

});
```

If parsing fails, plain text is returned, mainly to avoid a complete failure.